### PR TITLE
feat: MissingMetas and MissingOutput rules

### PR DESCRIPTION
**Pull request by a-frantz**
This pull request adds a new rule to `wdl-lint`.

- **Rule Name**: `MissingMetas` and `MissingOutput`

_Describe the rules you have implemented and link to any relevant issues._

Continuation of #93 

Before submitting this PR, please make sure:

- [x] You have added a few sentences describing the PR here.
- [x] You have added yourself or the appropriate individual as the assignee.
- [x] You have added at least one relevant code reviewer to the PR.
- [x] Your code builds clean without any errors or warnings.
- [x] You have added an entry to the relevant `CHANGELOG.md` (see
      ["keep a changelog"] for more information).
- [x] Your commit messages follow the [conventional commit] style.

Rule specific checks:

- [x] You have added the rule as an entry within `RULES.md`.
- [x] You have added the rule to the `rules()` function in `wdl-lint/src/v1.rs`.
- [x] You have added a test cases in `wdl-lint/tests/lints` that covers every 
      possible diagnostic emitted for the rule within the file where the rule  
      is implemented.
- [x] You have run `wdl-gauntlet --refresh` to ensure that there are no 
      unintended changes to the baseline configuration file (`Gauntlet.toml`).
- [x] You have run `wdl-gauntlet --refresh --arena` to ensure that all of the 
      rules added/removed are now reflected in the baseline configuration file 
      (`Arena.toml`).

[conventional commit]: https://www.conventionalcommits.org/en/v1.0.0/#summary
["keep a changelog"]: https://keepachangelog.com/en/1.0.0/

### CODE REVIEW

File: wdl-lint/src/v1/missing_metas.rs
Comment (peterhuene) [inline wdl-lint/src/v1/missing_metas.rs:131]: Nit: ever so slightly cheaper to check the boolean first.

```suggestion
        if inputs_present
            && task.metadata().next().is_none()
            && task.parameter_metadata().next().is_none()
```

File: wdl-lint/src/v1/missing_metas.rs
Comment (peterhuene) [inline wdl-lint/src/v1/missing_metas.rs:145]: Same as above:

```suggestion
        } else if inputs_present && task.parameter_metadata().next().is_none() {
```

File: wdl-lint/src/v1/missing_metas.rs
Comment (peterhuene) [inline wdl-lint/src/v1/missing_metas.rs:169]:
```suggestion
        if inputs_present
            && workflow.metadata().next().is_none()
            && workflow.parameter_metadata().next().is_none()
```

File: wdl-lint/src/v1/missing_metas.rs
Comment (peterhuene) [inline wdl-lint/src/v1/missing_metas.rs:183]:
```suggestion
        } else if inputs_present && workflow.parameter_metadata().next().is_none() {
```

File: wdl-lint/tests/lints/missing-meta-and-parameter_meta/source.wdl
Comment (peterhuene) [inline wdl-lint/tests/lints/missing-meta-and-parameter_meta/source.wdl:15]: We should remove the missing newline diagnostic in `source.errors` with a newline here.

**Review by peterhuene — COMMENTED** [review]
Looks good, just some relatively minor comments.

File: wdl-lint/src/v1/missing_metas.rs
Comment (peterhuene) [inline wdl-lint/src/v1/missing_metas.rs:159]: Copying @adthrasher's comment from the other PR:

> Can you import this to match TaskDefinition?

File: wdl-lint/src/v1/missing_output.rs
Comment (peterhuene) [inline wdl-lint/src/v1/missing_output.rs:103]: Here as well.

> Can you import this to match TaskDefinition?

```suggestion
        workflow: &WorkflowDefinition,
```

File: wdl-lint/src/v1/missing_metas.rs
Comment (peterhuene) [reply wdl-lint/src/v1/missing_metas.rs:159]:
```suggestion
        workflow: &WorkflowDefinition,
```

**Review by peterhuene — APPROVED** [review]
LGTM, but I think @adthrasher was also reviewing, so let's wait for his sign-off before merging.

File: wdl-lint/src/v1/missing_metas.rs
Comment (adthrasher) [inline wdl-lint/src/v1/missing_metas.rs:97]:
```suggestion
        "It is important that WDL code is well-documented. Every task and workflow should have both \
```

Just to avoid quoted quotes.

File: wdl-lint/tests/lints/command-mixed-line-cont/source.wdl
Comment (adthrasher) [inline wdl-lint/tests/lints/command-mixed-line-cont/source.wdl:5]: This is an interesting question that is probably out-of-scope. Long-term having these tests populated with additional fields that are irrelevant to the lint under test might complicate maintenance.

File: wdl-lint/tests/lints/missing-output/source.wdl
Comment (adthrasher) [inline wdl-lint/tests/lints/missing-output/source.wdl:4]: Does this not trip the missing `parameter_meta` rule also?

File: wdl-lint/tests/lints/command-mixed-line-cont/source.wdl
Comment (a-frantz) [reply wdl-lint/tests/lints/command-mixed-line-cont/source.wdl:5]: I had that thought while I was manually changing all these tests. Hopefully next week we will have working lint directives that we can use to simplify these test cases. I would have preferred to add `#@ expect: MissingMetas,MissingOutput` to all these.

File: wdl-lint/tests/lints/missing-output/source.wdl
Comment (a-frantz) [reply wdl-lint/tests/lints/missing-output/source.wdl:4]: There's no `input` so the `parameter_meta` check is skipped!

Comment (adthrasher) [reply wdl-lint/tests/lints/missing-output/source.wdl:4]: Thanks, missed that.

File: wdl-lint/tests/lints/command-mixed-line-cont/source.wdl
Comment (adthrasher) [reply wdl-lint/tests/lints/command-mixed-line-cont/source.wdl:5]: I suppose the number of rules that are applied to every document will be small. Most are "if `x` exists, check `x`" so those will skip.

**Review by adthrasher — APPROVED** [review]

**Pull request merged**
